app/controllers/oauth_controller.rb

```diff
@@ -40,21 +49,38 @@ def callback
     else
       # Catch the error and proceed to redirenct to '/' if we fail to get the access token
       begin
-        # Get the access token and other params with the authorization code, params[:code]
-        token = OAUTH_CLIENT.auth_code.get_token(params[:code], :redirect_uri => OAUTH_CALLBACK_URL)
-        session[:access_token] = token.token
-        session[:refresh_token] = token.refresh_token
+        # Get the access token and initialize it with ACCESS_TOKEN class
+        self.class.token = OAUTH_CLIENT.auth_code.get_token(params[:code], :redirect_uri => OAUTH_CALLBACK_URL)
+
+        session[:access_token] = self.class.token.token
+        session[:refresh_token] = self.class.token.refresh_token
 
         # Now we can put "Authorization: bearer #{session[:access_token]}" in the header when making an HTTP request
+        # Or, we can use self.class.token.get self.class.token.post to make requests
+
         # TODO: we need an authorization to check if the user is on the white list of the scsbuster
         redirect_to session[:original_url]
       rescue
+        puts 'Failed to get access token.'
```

Comment:
Can we take a pass through the app and replace `puts` statements with calls to `Rails.logger`?

Reply:
Sure!!! I didn't know the method is in Rails
So the syntax will be 
```
Rails.logger('Failed to get access token.')
```